version: '3'

services:

  redis:
    image: redis:7.0.2-alpine
    container_name: redis_app
    command: redis-server --save 20 1 --loglevel warning
    ports:
      - "6378:6379"
    volumes:
      - .data.cache:/data

  db:
    image: postgres
    user: postgres
    environment:
      - POSTGRES_PASSWORD=mysecretpassword
      - PGDATA=/var/lib/postgresql/data/pgdata
    volumes:
      - pg_data:/var/lib/postgresql/data/pgdata
    ports:
      - '5433:5432'
    healthcheck:
      test: ["CMD-SHELL", "pg_isready"]
      interval: 10s
      timeout: 5s
      retries: 5

  app:
    build: .
    tty: true
    command: python manage.py runserver 0.0.0.0:8000
    ports:
      - '8001:8000'
    depends_on:
      db:
        condition: service_healthy

  celery:
    restart: always
    build: .
    container_name: celery
    command: >
      bash -c "celery -A config worker --beat --loglevel=info"
    depends_on:
      - app
      - redis

volumes:
  pg_data:

